cvat-canvas/src/typescript/interactionHandler.ts

```diff
@@ -141,14 +147,15 @@ export class InteractionHandlerImpl implements InteractionHandler {
                         _e.preventDefault();
                         _e.stopPropagation();
                         self.remove();
+                        this.shapesWereUpdated = true;
+                        const shouldRiseEvent = this.shouldRaiseEvent(_e.ctrlKey);
                         this.interactionShapes = this.interactionShapes.filter(
                             (shape: SVG.Shape): boolean => shape !== self,
                         );
                         if (this.interactionData.startWithBox && this.interactionShapes.length === 1) {
                             this.interactionShapes[0].style({ visibility: '' });
                         }
-                        this.shapesWereUpdated = true;
-                        if (this.shouldRaiseEvent(_e.ctrlKey)) {
+                        if (shouldRiseEvent) {
```

Comment:
```suggestion
                        if (shouldRaiseEvent) {
```

Reply:
Removed

Reply:
Now I remember why I've added this variable. There was a bug in normal intelligent scissors mode. When we are removing all the anchors two black dots remain on canvas. Checkong shouldRaiseEvent before interaction data update solved the problem.